hudi-common/src/main/java/org/apache/hudi/common/table/log/HoodieLogFileReader.java

```diff
@@ -516,4 +521,23 @@ private static FSDataInputStream getFSDataInputStreamForGCS(FSDataInputStream fs
 
     return fsDataInputStream;
   }
+
+  /**
+   * Some filesystem(such as chdfs) will throw {@code IOException} instead of {@code EOFException}. It will cause error in isBlockCorrupted().
+   * Wrapped by {@code BoundedFsDataInputStream}, to check whether the desired offset is out of the file size in advance.
+   */
+  private static FSDataInputStream wrapStreamByBoundedFsDataInputStream(FileSystem fs,
```

Comment:
if we call this method in Line 490 above, we don't need lines 533 to 539 right. 
essentially
line 493 could be
```
return FSUtils.shouldWrappedByBoundedDataStream(fs) : new BoundedFsDataInputStream(fs, logFile.getPath(), fsDataInputStream): fsDataInputStream; 
```